R/flex_dashboard.R

```diff
@@ -263,6 +263,42 @@ flex_dashboard <- function(fig_width = 6.0,
     source_file <<- basename(input)
   }
 
+  #  special handling of 'global' chunk within runtime: shiny
+  if (!is.null(shiny::getDefaultReactiveDomain())) {
+
+    # options hook to (a) capture the chunk options for inspection below and
+    # (b) to ensure that 'global' chunks are not displayed in the document
+    chunk_options <- list()
+    knitr::opts_hooks$set(include = function(options) {
```

Comment:
I prefer leaving out this hook, and just let users type

    ```{global, include=FALSE}

Two reasons:

1) It will be confusing if they have

    ```{global, include=TRUE}

in which case I guess we need to respect their explicit choice of `include=TRUE`.

2) It seems one reason of this hook is to store chunk options in `chunk_options`, which may not be necessary -- you can use `knitr::opts_current$get('label')` below on line 283.